Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt

```diff
@@ -16,21 +16,98 @@
 
 package com.example.owl.ui
 
-import androidx.activity.OnBackPressedDispatcher
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.BottomNavigation
+import androidx.compose.material.BottomNavigationItem
+import androidx.compose.material.Icon
+import androidx.compose.material.LocalContentColor
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.material.primarySurface
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.CompositionLocalProvider
-import com.example.owl.ui.utils.LocalBackDispatcher
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+import androidx.navigation.compose.KEY_ROUTE
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.navigate
+import androidx.navigation.compose.rememberNavController
+import com.example.owl.ui.courses.CourseTabs
+import com.example.owl.ui.theme.BlueTheme
 import com.example.owl.ui.utils.ProvideImageLoader
 import com.google.accompanist.insets.ProvideWindowInsets
+import com.google.accompanist.insets.navigationBarsHeight
+import com.google.accompanist.insets.navigationBarsPadding
 
 @Composable
-fun OwlApp(backDispatcher: OnBackPressedDispatcher) {
+fun OwlApp(finishActivity: () -> Unit) {
+    ProvideWindowInsets {
+        ProvideImageLoader {
+            BlueTheme {
+                val tabs = CourseTabs.values()
+                val navController = rememberNavController()
+                OwlScaffold(
+                    bottomBar = { OwlBottomBar(navController = navController, tabs) }
+                ) { innerPaddingModifier ->
+                    NavGraph(
+                        finishActivity = finishActivity,
+                        navController = navController,
+                        modifier = innerPaddingModifier
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun OwlBottomBar(navController: NavController, tabs: Array<CourseTabs>) {
 
-    CompositionLocalProvider(LocalBackDispatcher provides backDispatcher) {
-        ProvideWindowInsets {
-            ProvideImageLoader {
-                NavGraph()
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    val currentRoute = navBackStackEntry?.arguments?.getString(KEY_ROUTE)
+        ?: CourseTabs.FEATURED.route
+
+    if (currentRoute in CourseTabs.values().map { it.route }) {
```

Comment:
nit: extract `CourseTabs.values().map { it.route }` to  const or `remember` or something. `values()` creates a new array every call, and then so will `map`

Reply:
Remembered the array and the map, although I suspect this is less performant overall :)

Reply:
Remembering probably is, but storing the end map in a companion object would be optimal.